Lexer FSM/main.py

```diff
@@ -6,39 +6,74 @@
 from Keywords import validate_keyword, keywords
 
 def main():
+    with open('./input.txt', 'r') as file:
+        # content contains the whole input file, you can access each character of the inputstring -> content[i]
+        content = file.read()
+
+    # given a string, will print the token type and lexeme
+    lexer(content)
+
     # Make instances of each finite state machine
     identifier_fsm = IdentifierFSM()
     integer_fsm = IntegerFSM()
     real_fsm = RealLexer()
 
-    endOfFile = False
-    # while not endOfFile:
-    #     lexer(string)
-
-    def lexer(string):
-        # Pass string to each FSM
-
-        if identifier_fsm.validate_identifier(string):
-            return (f"Token : Identifier, Lexeme: {string}")
-        elif integer_fsm.validate_integer(string):
-            return (f"Token : Integer, Lexeme: {string}")
-        elif real_fsm.validate_real(string):
-            return (f"Token : Real, Lexeme: {string}")
-        elif validate_keyword(string):
-            return (f"Token : Keyword, Lexeme: {string}")
-        elif validate_operator(string):
-            return (f"Token : Operator, Lexeme: {string}")
-        elif validate_separator(string):
-            return (f"Token : Separator, Lexeme: {string}")
-        else:
-            return "Invalid Token Type"
-
-    # Going to have to implement some logic that lets the lexer know how to separate operators and separators when there is no white space between them and other tokens
+    
+    def lexer(char):
+        lexeme = ""
+        inputCharTerminatesToken = False
+        for char in content:
+            
+            # if we are in one of the finite state machines and we get to a separator, then we should return a value we can identify from the validate_x functions
```

Comment:
Need to clarify here, is it just a separator, what about white space and operator?

Reply:
I get what you're saying though, I guess what would make sense is to have one way to tell if the actual validate_x is actually invalid and another way to see if the input char terminated the token

Reply:
> Need to clarify here, is it just a separator, what about white space and operator?

I think you're right and it should be a white space and operator as well.